src/extension/gogoduck/src/main/java/au/org/emii/gogoduck/worker/SubsetParameters.java

```diff
@@ -0,0 +1,47 @@
+package au.org.emii.gogoduck.worker;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import au.org.emii.gogoduck.worker.SubsetParameters.SubsetParameter;
+
+public class SubsetParameters extends HashMap<String, SubsetParameter> {
```

Comment:
Does this class itself need to be thread-safe?  If I'm reading things correctly, I think not, as it's the one thread doing the writing, at the start, before the multiple threads simply *read* from here.